### DISCUSSION TITLE
Build Time is huge after gatsby V3 and [gatsby-plugin-image]

### DISCUSSION THREAD

**Discussion by ahmadkhalaf1** [gatsbyjs/gatsby#30201] [Help]
Hi , 
so after updating my project to gatsby v3 and gatsby-plugin-image , my build time locally jumped from around 9min to 40 minute !!! 
i have no idea why , the code is the same , and its even cleaner with v3 and new image plugin , but the build is taking huge time , how can i know why and what can i do to fix it please its affecting my project on production .

thanks

**Comment by ahmadkhalaf1** [comment]
so seems like the issue is not because of V3 , but more about gatsby-plugin-image , 
after some hours of testing , i removed the new image plugin , and only applied it to 1 hero image in my 350 blog pages , and when i build the project it take huge time at 
onPostBootstrap Step ,
```
this is how i use my query 

   imgUrl {
                  childImageSharp {
                    gatsbyImageData(
                      breakpoints: [576, 992, 1200, 1300]
                      placeholder: TRACED_SVG
                      layout: FULL_WIDTH
                      formats: [AUTO, WEBP, AVIF]
                    )
                  }
                }
            
                
 then 
 <GatsbyImage
loading="eager"
 objectPosition="50% 60%"
   image={getImage(data.imgUrl) || ({} as IGatsbyImageData)}
   alt="Header"
    />
      
 am i doing something wrong?
 
 this is my previous query with old image plugin 
 

 imgUrl {
                  childImageSharp {
                    fluid(
                      maxWidth: 1400
                      maxHeight: 560
                      srcSetBreakpoints: [576, 992, 1200, 1300]
                    ) {
                      tracedSVG
                      aspectRatio
                      sizes
                      src
                      srcSet
                      srcSetWebp
                      srcWebp
                    }
                  }
                }
```

**Reply by KyleAMathews** [reply]
The main difference it seems is you added `AVIF`. Try removing that to see if the build time is similar

**Reply by ahmadkhalaf1** [reply]
> The main difference it seems is you added `AVIF`. Try removing that to see if the build time is similar

good catch !! after i removed AVIF in the project build time is now again 9 minutes ! 
anyidea how we can fix this ( if this is an issue ) . 
Thanks

**Reply by KyleAMathews** [reply]
I'm not entirely sure why AVIF is so much more expensive to process. Probably some combination of it's a new library with less optimizations + it's a more complex algorithm than JPEG.

You can setup your config to disable AVIF & WebP locally and then enable them for production builds on CI. [Gatsby Cloud](https://www.gatsbyjs.com/cloud/) automatically caches image processing so it doesn't impact your build time on subsequent builds.
